src/main/java/wooteco/subway/line/LineService.java

```diff
@@ -0,0 +1,79 @@
+package wooteco.subway.line;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import wooteco.subway.section.SectionDao;
+import wooteco.subway.station.Station;
+import wooteco.subway.station.StationDao;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class LineService {
+
+    private final LineDao lineDao;
+    private final StationDao stationDao;
+    private final SectionDao sectionDao;
+
+    @Autowired
+    public LineService(LineDao lineDao, StationDao stationDao, SectionDao sectionDao) {
+        this.lineDao = lineDao;
+        this.stationDao = stationDao;
+        this.sectionDao = sectionDao;
+    }
+
+    public LineResponse createLine(long upStationId, long downStationId, String lineName, String lineColor) {
+        final Optional<Line> lineWithSameName = lineDao.findByName(lineName);
+        if (lineWithSameName.isPresent()) {
+            throw new IllegalArgumentException("노선 이름이 중복됩니다.");
+        }
+        Line line = lineDao.save(lineName, lineColor);
+        sectionDao.save(line.getId(), upStationId, downStationId);
+        return LineResponse.from(line);
+    }
+
+    public List<LineResponse> showLines() {
+        final List<Line> lines = lineDao.findAll();
+        return lines.stream()
+                .map(LineResponse::from)
+                .collect(Collectors.toList());
+    }
+
+    public LineResponse showLine(long lineId) {
+        final Line line = findLineById(lineId);
+        final List<Station> stationsInLine = findStationsInLine(lineId);
+        line.setStations(stationsInLine);
+        return LineResponse.from(line);
+    }
+
+    private Line findLineById(long lineId) {
+        final Optional<Line> lineFoundById = lineDao.findById(lineId);
+        if (!lineFoundById.isPresent()) {
+            throw new IllegalArgumentException("해당 id에 대응하는 노선이 없습니다.");
+        }
+        return lineFoundById.get();
+    }
```

Comment:
```suggestion
        return lineDao.findById(lineId).orElseThrow(() -> new IllegalArgumentException(lineId + "에 해당하는.."));
```
요렇게 한줄로 하면 가독이 더 좋아질거에요~ 다른 optional 처리쪽도 마찬가지입니다